src/hooks/useQuery.ts

```diff
@@ -0,0 +1,72 @@
+import { queryApi } from "../client";
+import { DocumentNode, GraphQLFormattedError } from "graphql";
+import { useState, useEffect, useCallback, useRef } from "react";
+import { useExtension } from "./ExtensionProvider";
+
+interface BackgroundQueryOptions<V> {
+  variables?: V;
+  skip?: Boolean;
+}
+
+type QueryError = GraphQLFormattedError | Error;
+
+export const useQuery = <T, V>(
+  query: DocumentNode,
+  { skip, variables }: BackgroundQueryOptions<V> = {}
+) => {
+  const { id } = useExtension();
+  const mounted = useRef(true);
+  const [data, setData] = useState<T | null>(null);
+  const [error, setError] = useState<QueryError | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const makeQuery = useCallback(
+    async (argVars?: V) => {
+      try {
+        const resp = await queryApi<T, V>(
+          query,
+          // @ts-ignore variables are kinda weird
```

Comment:
So this is like `eslint-disable`?

Reply:
yep, need to figure out how I can type this better, right now the tricky part is making it optional if there is no variables. Going to need to revise this in the future but with this, it allows for the behavior wanted.